### PR TITLE
chore(button): convert to use Box

### PR DESCRIPTION
**Pull request by richbachman**
Button now uses Box under the hood

### CODE REVIEW

File: packages/paste-core/primitives/box/src/index.tsx [hunk 1]
```diff
@@ -77,37 +78,41 @@ export interface BoxStyleProps
...
+  _even?: BoxBaseStyleProps;
+  _visited?: BoxBaseStyleProps;
+  _readOnly?: BoxBaseStyleProps;
+  _first?: BoxBaseStyleProps;
+  _last?: BoxBaseStyleProps;
+  _groupHover?: BoxBaseStyleProps;
+  _notFirst?: BoxBaseStyleProps;
+  _notLast?: BoxBaseStyleProps;
+  _placeholder?: BoxBaseStyleProps;
+  __moz_focus_inner?: BoxBaseStyleProps;
```
Comment (TheSisb) [inline packages/paste-core/primitives/box/src/index.tsx:109]: Added this as it was being used in CSS directly before.

File: packages/paste-core/components/button/src/DestructiveButton.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+// import * as PropTypes from 'prop-types';
```
Comment (richbachman) [inline packages/paste-core/components/button/src/DestructiveButton.tsx:2]: Do we need PropTypes on these individual components? If not, we can remove this comment.

File: packages/paste-core/components/button/src/LinkButton.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+// import * as PropTypes from 'prop-types';
```
Comment (richbachman) [inline packages/paste-core/components/button/src/LinkButton.tsx:2]: Same. Either remove or add PropTypes if needed.

File: packages/paste-core/components/button/src/PrimaryButton.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+// import * as PropTypes from 'prop-types';
```
Comment (richbachman) [inline packages/paste-core/components/button/src/PrimaryButton.tsx:2]: Same. Either remove or add PropTypes if needed.

File: packages/paste-core/components/button/src/ResetButton.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+// import * as PropTypes from 'prop-types';
```
Comment (richbachman) [inline packages/paste-core/components/button/src/ResetButton.tsx:2]: Same. Either remove or add PropTypes if needed.

File: packages/paste-core/components/button/src/SecondaryButton.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+// import * as PropTypes from 'prop-types';
```
Comment (richbachman) [inline packages/paste-core/components/button/src/SecondaryButton.tsx:2]: Same. Either remove or add PropTypes if needed.

File: packages/paste-core/layout/stack/src/index.tsx [hunk 7]
```diff
@@ -19,7 +19,7 @@ type StackOrientationOptions = 'horizontal' | 'vertical';
 export type StackOrientation = ResponsiveValue<StackOrientationOptions>;
 
 interface StackStyleProps extends Pick<LayoutProps, 'display'>, Pick<FlexboxProps, 'alignItems' | 'flexWrap'> {}
-export interface StackProps extends Pick<BoxProps, 'as'>, React.HTMLAttributes<any> {
+export interface StackProps extends BoxProps {
```
Comment (SiTaggart) [inline packages/paste-core/layout/stack/src/index.tsx:22]: Extending `BoxProps` here isn't necessarily true as we `safelySpreadBoxProps`, removing style props passed down. You probably want `BoxElementProps` instead.

File: packages/paste-core/primitives/box/src/index.tsx (hunk 1 above)
Comment (SiTaggart) [reply packages/paste-core/primitives/box/src/index.tsx:109]: You'll need to add this to here https://github.com/twilio-labs/paste/blob/main/packages/paste-core/primitives/box/src/PseudoPropStyles.ts

Comment (SiTaggart) [reply packages/paste-core/primitives/box/src/index.tsx:109]: In fact I think if you remove `: {[key: string]: string}` from `PseudoPropStyles` in https://github.com/twilio-labs/paste/blob/main/packages/paste-core/primitives/box/src/PseudoPropStyles.ts You might be able to dynamically create this interface like 

```
type PseudoSelectors = keyof typeof PseudoPropStyles;
type PseudoStylesProps = {
  [key in PseudoSelectors]: BoxStyleProps;
}```

so we don't have to keep these in sync

File: packages/paste-core/layout/stack/src/index.tsx (hunk 7 above)
Comment (TheSisb) [reply packages/paste-core/layout/stack/src/index.tsx:22]: 🧠

File: packages/paste-core/primitives/box/src/index.tsx (hunk 1 above)
Comment (TheSisb) [reply packages/paste-core/primitives/box/src/index.tsx]: Good idea, fixed!

**Pull request merged**